examples/erc721-consecutive/tests/erc721-consecutive.rs

```diff
@@ -10,52 +14,58 @@ use crate::abi::Erc721;
 
 mod abi;
 
+sol!("src/constructor.sol");
+
 fn random_token_id() -> U256 {
     let num: u32 = rand::random();
     U256::from(num)
 }
 
-async fn deploy(rpc_url: &str, private_key: &str) -> eyre::Result<Address> {
-    e2e::deploy(rpc_url, private_key, None).await
+async fn deploy(
+    rpc_url: &str,
+    private_key: &str,
+    receivers: Vec<Address>,
+    amounts: Vec<u128>,
+) -> eyre::Result<Address> {
+    let args = Erc721ConsecutiveExample::constructorCall { receivers, amounts };
+    let args = alloy::hex::encode(args.abi_encode());
+    e2e::deploy(rpc_url, private_key, Some(args)).await
 }
 
 #[e2e::test]
 async fn constructs(alice: Account) -> eyre::Result<()> {
-    let contract_addr = deploy(alice.url(), &alice.pk()).await?;
-    let contract = Erc721::new(contract_addr, &alice.wallet);
-
     let alice_addr = alice.address();
     let receivers = vec![alice_addr];
-    let amounts = vec![uint!(10_U256)];
-    let _ = watch!(contract.init(receivers, amounts))?;
+    let amounts = vec![10_u128];
+    let contract_addr =
+        deploy(alice.url(), &alice.pk(), receivers, amounts).await?;
+    let contract = Erc721::new(contract_addr, &alice.wallet);
+
     let balance = contract.balanceOf(alice_addr).call().await?.balance;
     assert_eq!(balance, uint!(10_U256));
     Ok(())
 }
 
 #[e2e::test]
 async fn mints(alice: Account) -> eyre::Result<()> {
-    let contract_addr = deploy(alice.url(), &alice.pk()).await?;
+    let batch_size = 10_u128;
+    let receivers = vec![alice.address()];
+    let amounts = vec![batch_size];
+    let contract_addr =
+        deploy(alice.url(), &alice.pk(), receivers, amounts).await?;
     let contract = Erc721::new(contract_addr, &alice.wallet);
 
-    let Erc721::balanceOfReturn { balance: initial_balance } =
-        contract.balanceOf(alice.address()).call().await?;
-
-    let batch_size = uint!(10_U256);
-    let receipt =
-        receipt!(contract.init(vec![alice.address()], vec![batch_size]))?;
-    let from_token_id = U256::from(Params::FIRST_CONSECUTIVE_ID);
-    let to_token_id = from_token_id + batch_size - uint!(1_U256);
-    assert!(receipt.emits(Erc721::ConsecutiveTransfer {
-        fromTokenId: from_token_id,
-        toTokenId: to_token_id,
-        fromAddress: Address::ZERO,
-        toAddress: alice.address(),
-    }));
+    // TODO#q: check for ConsecutiveTransfer transfer event after deploy.
```

Comment:
Checking for emitted events for construction will be possible after koba being able to return transaction receipt